### DISCUSSION TITLE
QT6 and SavvyCAN

### DISCUSSION THREAD

**Discussion by collin80** [collin80/SavvyCAN#705] [General]
Various people have submitted patches to try to get the codebase better positioned to work with QT6. QT6 has now been out approximately 3 years and so I think it's time to get full QT6 support and to start releasing the binary builds as QT6 binaries. There is (and has been since May 2022) a QT6 Development Build. As of right now, the QT6 build includes all the commits from master plus extra commits to make things work properly with QT6. It should more or less kind of/sort of nearly almost appear to be some manner of facsimile of working. But, there are probably still some bugs and there are more things I'd like to change to make the code "more proper" and not crash. If you're brave, you can give the QT6 version a try then report back here all the stuff that doesn't actually work right. Things probably will continue to be slow in terms of my ability to fix bugs and merge commits until I get QT6 support solidified. But, it's got to happen eventually as QT5 support has already officially ended for open source projects like this. Ordinarily that wouldn't be so awful but it means there are no bug fixes for QT5 any longer. And, QT5 does not really play that nicely with hiDPI monitors which more and more of us have. So, QT6 is the path forward and this is just a note saying I am working on it and there should be good progress in the near future. It already is semi-functional. Thanks  to everyone who submitted patches to help getting it ported over.

TL;DR - Development is transitioning to QT6 as QT5 is deprecated and will have no more updates of any kind.

**Comment by EngineerTunerNorway** [comment]
Oh yeah. It feels so mutch better, faster....

Good work.

**Comment by daveHitchman** [comment]
The build instructions say to use 5.14.x or newer and even, very nicely, points me at qt.io to get it. Then the fun starts ... a myriad of options, a form to fill in, an instruction to enter a password that doesnt appear to ever arrive. I know QT is a useful framework but it appears that QT is no longer simply downloadable to build packages like this. Does anyone have a solution?

**Reply by EngineerTunerNorway** [reply]
I can upload it to a  server. Gimmi 30 min ill fix it.

**Reply by EngineerTunerNorway** [reply]
QT6: https://drive.google.com/file/d/1yFU3FC8vEPfM04y8ZQjm0CT973NmfOMS/view?usp=sharing
QT5.15.2-src: https://drive.google.com/file/d/1FRi5VMTKGtInwBKgS8Cu2fzEmFMmNa1S/view?usp=sharing

**Reply by daveHitchman** [reply]
Fantastic, downloaded, with a quick fix for the CRs (for anyone else on linux sed -i -e 's/\r$//' configure) qt base has built, so I will now try the rest. Thanks a lot for your help

**Reply by EngineerTunerNorway** [reply]
no problem.

**Reply by daveHitchman** [reply]
Sadly it didnt work, I got errors in CanConnection about loadRelaxed - as I dont want to actually connect to a can box just analyse already collected logs I just bodged that. Then dbchandler moaned about swapItemsAt which was hackable with a change to swap... which I guess indicates a wrong QT version, then dbcsignaleditor cant cope with textActtivated. At this point I abandoned the download from you and tried to follow QTs instructions and download, that just plain didnt work either. This seems a lot of work for zero forward progress which seems sad after the effort so many people have put into this.

**Reply by douglasheld2** [reply]
Did you get errors building qt5 or qt6?

**Comment by douglasheld2** [comment]
I would just like to report I've built SavvyCAN from the continuous-QT6 today against Qt6.7.2. Soo... that's working. Thanks @collin80 

My HOWTO is here: https://github.com/collin80/SavvyCAN/discussions/804
